core/string/ustring.cpp

```diff
@@ -1644,6 +1644,34 @@ String String::hex_encode_buffer(const uint8_t *p_buffer, int p_len) {
 	return ret;
 }
 
+Vector<uint8_t> String::hex_decode() const {
+	const String *s = this;
+	Vector<uint8_t> out;
+	int len = s->size() / 2;
+	out.resize(len);
+	for (int i = 0; i < len; i++) {
+		String part = s->substr(i * 2, 2);
```

Comment:
You create a substring for each pair of characters, but you can just get those characters.

Reply:
yeah i went to sleep thinking about optimizing this :>